### PR TITLE
Distribution Independent benchmark level 2 controls

### PR DESCRIPTION
**Pull request by ajegher-sn**
|ID|Check|Status|Notes|
|-|-|-|-|
|1.1.1.8|Ensure mounting of FAT filesystems is limited|Not Implemented|Two different checks whether the system is UEFI or not|
|1.1.6|Ensure separate partition exists for /var|Implemented||
|1.1.7|Ensure separate partition exists for /var/tmp|Implemented||
|1.1.11|Ensure separate partition exists for /var/log|Implemented||
|1.1.12|Ensure separate partition exists for /var/log/audit|Implemented||
|1.1.13|Ensure separate partition exists for /home|Implemented||
|1.1.22|Disable Automounting|Implemented|Already implemented, was server level 1|
|1.1.23|Disable USB Storage|Implemented|Already implemented, was server level 1|
|1.6.1.1|Ensure SELinux or AppArmor are installed|Implemented||
|1.6.2.1|Ensure SELinux is not disabled in bootloader configuration|Implemented||
|1.6.2.2|Ensure the SELinux state is enforcing|Implemented||
|1.6.2.3|Ensure SELinux policy is configured|Implemented||
|1.6.2.4|Ensure SETroubleshoot is not installed|Implemented||
|1.6.2.5|Ensure the MCS Translation Service (mcstrans) is not installed|Implemented||
|1.6.2.6|Ensure no unconfined daemons exist|Implemented||
|1.6.3.1|Ensure AppArmor is not disabled in bootloader configuration|Implemented||
|1.6.3.2|Ensure all AppArmor Profiles are enforcing|Implemented with limitations|Can only check if a profile is in complain mode, and not if it was never loaded or removed|
|2.2.4|Ensure CUPS is not enabled|Implemented|Already implemented, was server level 1|
|3.4.1|Ensure DCCP is disabled|Implemented||
|3.4.2|Ensure SCTP is disabled|Implemented||
|3.4.3|Ensure RDS is disabled|Implemented||
|3.4.4|Ensure TIPC is disabled|Implemented||
|3.6|Ensure wireless interfaces are disabled|Implemented||
|3.7|Disable IPv6|Implemented||
|4.1.1.1|Ensure audit log storage size is configured|Implemented||
|4.1.1.2|Ensure system is disabled when audit logs are full|Implemented||
|4.1.1.3|Ensure audit logs are not automatically deleted|Implemented||
|4.1.2|Ensure auditd is installed|Implemented||
|4.1.3|Ensure auditd service is enabled|Implemented||
|4.1.4|Ensure auditing for processes that start prior to auditd is enabled|Implemented||
|4.1.5|Ensure events that modify date and time information are collected|Implemented||
|4.1.6|Ensure events that modify user/group information are collected|Implemented||
|4.1.7|Ensure events that modify the system's network environment are|Implemented||
|4.1.8|Ensure events that modify the system's Mandatory Access|Implemented||
|4.1.9|Ensure login and logout events are collected|Implemented||
|4.1.10|Ensure session initiation information is collected|Implemented||
|4.1.11|Ensure discretionary access control permission modification|Implemented||
|4.1.12|Ensure unsuccessful unauthorized file access attempts are collected|Implemented||
|4.1.13|Ensure use of privileged commands is collected|Not Implemented|Requires multi-step actions depending on mounted partitions and SUID/SGID files|
|4.1.14|Ensure successful file system mounts are collected|Implemented||
|4.1.15|Ensure file deletion events by users are collected|Implemented||
|4.1.16|Ensure changes to system administration scope (sudoers) is collected|Implemented||
|4.1.17|Ensure system administrator actions (sudolog) are collected|Implemented||
|4.1.18|Ensure kernel module loading and unloading is collected|Implemented||
|4.1.19|Ensure the audit configuration is immutable|Implemented||
|5.2.6|Ensure SSH X11 forwarding is disabled|Implemented||
|5.2.21|Ensure SSH AllowTcpForwarding is disabled|Implemented||
|5.4.5|Ensure default user shell timeout is 900 seconds or less|Implemented||
|6.1.1|Audit system file permissions|Not Implemented|Requires parsing the dpkg/rpm local database|

### CODE REVIEW

File: configs/defs/fallback.textproto [hunk 1]
```diff
@@ -10187,3 +10187,2726 @@ benchmark_configs: {
...
+    cis_benchmark: {
+      profile_level: 2
+      severity: LOW
+    }
+    scan_instructions:
+      "scan_type_specific:{"
+      " instance_scanning: {"
+      "  check_alternatives:{"
+      "    file_checks:{"
+      "      files_to_check:{"
```
[the remark this answers is not in the document]
Comment (ajegher-sn) [reply configs/defs/fallback.textproto:10686]: Can I use that to go through all the running pids? From what I understood this benchmark should check all the daemons that are not defined in SELinux policy

[the remark this answers is not in the document]
Comment (erikvarga) [reply configs/defs/fallback.textproto]: Yeah, it goes through all the pid entries in /proc. You can see the impl here: https://github.com/google/localtoast/blob/master/scannerlib/fileset/file_set.go#L225

There should be some examples in previous configs on how to use the field.

[the remark this answers is not in the document]
Comment (ajegher-sn) [reply configs/defs/fallback.textproto]: Maybe I am missing something, from the [implementation](https://github.com/google/localtoast/blob/39da8e12ba79b8ccf46cd762e9afe29bf96b900f/scannerlib/fileset/file_set.go#L255C6-L255C14) it looks like it requires a specific `proc_name` to work and I don't have it in this benchmark.

File: configs/defs/fallback.textproto [hunk 2]
```diff
@@ -5491,140 +5701,1902 @@ benchmark_configs: {
...
+      "```max_log_file_action = keep_logs```"
+    cis_benchmark: {
+      profile_level: 2
+      severity: LOW
+    }
+    scan_instructions:
+    "generic:{"
+      "      check_alternatives:{"
+      "        file_checks:{"
+      "          files_to_check:{single_file:{path:\"/etc/audit/auditd.conf\"}}"
```
Comment (erikvarga) [inline configs/defs/fallback.textproto:6330]: Add check_alternative for file not existing

File: configs/defs/fallback.textproto [hunk 3]
```diff
@@ -5491,140 +5701,1902 @@ benchmark_configs: {
...
+      "Notes: Additional methods of enabling a service exist. "
+      "Consult your distribution documentation for appropriate methods."
+    cis_benchmark: {
+      profile_level: 2
+      severity: LOW
+    }
+    scan_instructions:
+      "generic:{"
+      "  check_alternatives:{"
+      "    file_checks:{"
```
Comment (erikvarga) [inline configs/defs/fallback.textproto:6415]: Add check_alternative for /etc/systemd/system/ not existing

File: configs/defs/fallback.textproto [hunk 4]
```diff
@@ -5491,140 +5701,1902 @@ benchmark_configs: {
...
+      "the appropriate grub configuration file for your environment."
+    cis_benchmark: {
+      profile_level: 2
+      severity: LOW
+    }
+    scan_instructions:
+      "generic:{"
+      " check_alternatives:{"
+      "    file_checks:{"
+      "      files_to_check:{single_file:{path:\"/boot/grub2/grub.cfg\"}}"
```
Comment (erikvarga) [inline configs/defs/fallback.textproto:6460]: Add check_alternative for e.g. /boot not existing

File: configs/defs/fallback.textproto [hunk 5]
```diff
@@ -5491,140 +5701,1902 @@ benchmark_configs: {
...
+    cis_benchmark: {
+      profile_level: 2
+      severity: LOW
+    }
+    scan_instructions:
+      "scan_type_specific:{"
+      "    instance_scanning:{"
+      "        check_alternatives:{"
+      "            file_checks:{"
+      "                files_to_check:{single_file:{path: \"/etc/audit/audit.rules\"}}"
```
Comment (erikvarga) [inline configs/defs/fallback.textproto:6537]: There are a couple more checks here and below where we'd fail if it ran on a binary-only container - let's add check_alternatives for the base files/dirs not existing

File: configs/defs/fallback.textproto (hunk 1 above)
[the remark this answers is not in the document]
Comment (SN-Eros) [reply configs/defs/fallback.textproto]: the expected path is `/proc/[PID]/attr/current`, which is documented here: https://www.kernel.org/doc/Documentation/ABI/testing/procfs-attr-current

[the remark this answers is not in the document]
Comment (erikvarga) [reply configs/defs/fallback.textproto]: Oh, that makes more sense. In that case can you add a comment that you're looking at /proc/[PID]/attr/current? It wasn't obvious from the check instruction.

File: configs/defs/fallback.textproto [hunk 6]
```diff
@@ -6477,6 +8485,55 @@ benchmark_configs: {
...
+      "Edit the /etc/ssh/sshd_config file to set the parameter as follows: \n"
+      "```X11Forwarding no```"
+    cis_benchmark: {
+      profile_level: 2
+      severity: LOW
+    }
+    scan_instructions:
+      "generic:{check_alternatives:{"
+      "  file_checks:{"
+      "    files_to_check:{single_file:{path:\"/etc/ssh/sshd_config\"}}"
```
Comment (erikvarga) [inline configs/defs/fallback.textproto:8512]: Add a check alternative for /etc/ssh/sshd_config not existing - also for the checks below

File: configs/defs/fallback.textproto [hunk 7]
```diff
@@ -5491,140 +5701,1938 @@ benchmark_configs: {
...
+    cis_benchmark: {
+      profile_level: 2
+      severity: LOW
+    }
+    scan_instructions:
+      "scan_type_specific:{"
+      "    instance_scanning:{"
+      "        check_alternatives:{"
+      "            file_checks:{"
+      "                files_to_check:{single_file:{path: \"/etc/audit/audit.rules\"}}"
```
Comment (erikvarga) [inline configs/defs/fallback.textproto:6573]: Add a check alternative for/etc/audit not existing - also for the checks below

File: configs/defs/fallback.textproto [hunk 8]
```diff
@@ -8159,24 +10269,71 @@ benchmark_configs: {
...
       "          group_index: 1"
-      "          type: NO_LESS_RESTRICTIVE_UMASK"
-      "          const: 027"
+      "          type: LESS_THAN"
+      "          const: 901"
       "        }"
       "      }"
       "    }"
       "  }"
-      "}"
```
Comment (erikvarga) [inline configs/defs/fallback.textproto:8172]: Keep the check alternatives for /etc/bash/bashrc (etc.) not existing

File: configs/defs/fallback.textproto [hunk 9]
```diff
@@ -10187,3 +12344,695 @@ benchmark_configs: {
...
+      "# apt-get install libselinux1\n"
+      "# zypper install libselinux```\n"
+      "The previous commands install SELinux, use the appropriate package if AppArmor is desired. "
+    cis_benchmark: {
+      profile_level: 2
+      severity: LOW
+    }
+    scan_instructions:
+      "generic:{"
+        "check_alternatives:{"
```
Comment (erikvarga) [inline configs/defs/fallback.textproto:12369]: check_alternative for /usr/bin and /usr/sbin not existing

File: configs/defs/fallback.textproto [hunk 10]
```diff
@@ -10187,3 +12344,695 @@ benchmark_configs: {
...
+      "```SELINUX=enforcing```"
+    cis_benchmark: {
+      profile_level: 2
+      severity: LOW
+    }
+    scan_instructions:
+      "generic:{"
+      " check_alternatives:{"
+      "    file_checks:{"
+      "      files_to_check:{single_file:{path:\"/etc/selinux/config\"}}"
```
Comment (erikvarga) [inline configs/defs/fallback.textproto:12496]: check_alternative for /etc/selinux/config not existing, here and below

File: configs/defs/fallback.textproto [hunk 11]
```diff
@@ -10187,3 +12373,710 @@ benchmark_configs: {
...
+      "```SELINUXTYPE=targeted```\n"
+      "Notes: If your organization requires stricter policies, "
+      "ensure that they are set in the /etc/selinux/config file."
+    cis_benchmark: {
+      profile_level: 2
+      severity: LOW
+    }
+    scan_instructions:
+      "generic:{"
+      " check_alternatives:{"
```
Comment (erikvarga) [inline configs/defs/fallback.textproto:12575]: check_alternative for /etc/selinux/config not existing here too

Comment (erikvarga) [reply configs/defs/fallback.textproto:12575]: You can verify if all use cases are covered by running Localtoast with chroot set to an empty directory with e.g. `./localtoast [...] --chroot=/tmp/dir` and checking that no benchmarks are non-compliant

File: configs/defs/fallback.textproto (hunk 3 above)
Comment (erikvarga) [reply configs/defs/fallback.textproto:6445]: This check_alternative is still missing - if I run Localtoast chrooted to an empty dir I get two failed benchmarks, `selinux-policy-configured-fallback` and `auditd-service-enabled-fallback`

**Pull request merged**